### PR TITLE
code for benchmark

### PR DESCRIPTION
**Pull request by jiahuanluo**
code for paper **Real_World_Image_Datasets_for_Federated_Learning**

### CODE REVIEW

File: research/federated_object_detection_benchmark/experiments/log_formatter.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        elif "server_test_loss" in line:
+            server_test_loss.append(round(float(line.split(" ")[-1]), 4))
+        elif "server_test_map" in line:
+            server_test_map.append(round(float(line.split(" ")[-1]), 4))
+        elif "server_test_recall" in line:
+            server_test_recall.append(round(float(line.split(" ")[-1]), 4))
+    output.write("round,train_loss,test_map,test_recall\n")
+    for r, loss, mAP, recall in zip(round_, train_loss, server_test_map, server_test_recall):
+        output.write("{},{},{},{}\n".format(r, loss, mAP, recall))
+    # x = np.array(round_)
```
Comment (yankang18) [inline research/federated_object_detection_benchmark/experiments/log_formatter.py]: remove canceled comments. likewise in other code documents

File: research/federated_object_detection_benchmark/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Federated-Benchmark: A Benchmark of Real-world Images Dataset for Federated Learning
+
+## Overview
+We present a real-world image dataset, reflecting the characteristic real-world federated learning scenarios, and provide provided an extensive benchmark on model performance, efficiency, and communication in a federated learning setting.
```
[the remark this answers is not in the document]
Comment (jiahuanluo) [reply research/federated_object_detection_benchmark/README.md]: fixed

File: research/federated_object_detection_benchmark/experiments/log_formatter.py (hunk 1 above)
Comment (jiahuanluo) [reply research/federated_object_detection_benchmark/experiments/log_formatter.py]: fixed

File: research/federated_object_detection_benchmark/data/voc_dataset.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    def __init__(self, data_dir, label_names, split='train',
+                 use_difficult=False, return_difficult=False):
+
+        # if split not in ['train', 'trainval', 'val']:
+        #     if not (split == 'test' and year == '2007'):
+        #         warnings.warn(
+        #             'please pick split from \'train\', \'trainval\', \'val\''
+        #             'for 2012 dataset. For 2007 dataset, you can pick \'test\''
+        #             ' in addition to the above mentioned splits.'
+        #         )
```
[the remark this answers is not in the document]
Comment (jiahuanluo) [reply research/federated_object_detection_benchmark/data/voc_dataset.py]: necessary

**Pull request merged**